torch/linalg/__init__.py

```diff
@@ -1807,6 +1816,82 @@
     https://en.wikipedia.org/wiki/Invertible_matrix#The_invertible_matrix_theorem
 """)
 
+solve_triangular = _add_docstr(_linalg.linalg_solve_triangular, r"""
+linalg.solve_triangular(A, B, *, left=True, upper=True, unitriangular=False, out=None) -> Tensor
+
+Computes the solution of a triangular system of linear equations with a unique solution.
+
+Letting :math:`\mathbb{K}` be :math:`\mathbb{R}` or :math:`\mathbb{C}`,
+this function computes the solution :math:`X \in \mathbb{K}^{n \times k}` of the **linear system**
+associated to the upper triangular matrix :math:`A \in \mathbb{K}^{n \times n}` without zeros on the diagonal
+(that is, it is `invertible`_) and the rectangular matrix , :math:`B \in \mathbb{K}^{n \times k}`,
+which is defined as
+
+.. math:: AX = B
+
+If :attr:`upper`\ `= False`, :math:`A` will be assumed to be lower triangular without zeros on the diagonal.
+
+If :attr:`left`\ `= False`, the returned matrix :math:`X \in \mathbb{K}^{n \times k}` will solve the system
+
+.. math::
+
+    XA = B\mathrlap{\qquad A \in \mathbb{K}^{k \times k}, B \in \mathbb{K}^{n \times k}.}
```

Comment:
I find this name very confusing. What about computing the solution in the column vs the row space?

Reply:
I think the columns vs rows is too technical. I don't think `left` is perfect, but it gets the job done. Note that the only thing that's important in the problem is `A`, as we can think of this as `AX = B` or `X = A^{-1}B`, and in both cases `A` is acting on the left.
Happy to hear other suggestions though. What do you think @IvanYashchuk ?

P.S. About the `transpose` parameter, note that this logic removes it, which is nice as it was too confusing. Now to transpose you would just sent the matrix transposed on PyTorch, which is more orthogonal with the rest of the PyTorch API.

Reply:
I'm in favor of the `left` argument. It's simple on the user side as well as on the backend side. I think the description in the main text and in the "Args" section does a good job so that it's not ambiguous.
As Mario already mentioned: If `left=True` - solve for `X` the equation`AX=B` or alternatively compute`X=A⁻¹ B`; if `left=False` - solve `XA=B` or compute `X=BA⁻¹`, the `left` argument describes the position of matrix `A` or its inverse in both formulations.
Also left or right matrix divisions in the literature refer to the position of the inverse matrix in a matrix multiplication operation (maybe it's coming from Matlab, but still the terms are common).